runtime/onert/core/src/exec/DynamicShapeInferer.cc

```diff
@@ -48,12 +48,12 @@ void DynamicShapeInferer::handleBinaryArithmeticOp(const ir::Operation &op,
 
     So, only when all inputs are static, we can skip dynamic shape inference.
   */
-  if ((!lhs->is_dynamic()) && (!rhs->is_dynamic()))
-    return;
-
   auto output_idx = op.getOutputs().at(0);
   auto output = _tensor_registry->getITensor(output_idx);
 
+  if ((!lhs->is_dynamic()) && (!rhs->is_dynamic()) && !output->is_dynamic())
```

Comment:
To make this code more readable, could you define a function like this?
```
// in case of output tensor of an op, it is possible that 
// the output became dynamic although it had been static before.
// Once a tensor becomes dynamic, it will lost memory allocated for static.
// Therefore once output is dynamic, it should be treated as dynamic tensor. (memory should be allocated at runtime)
// `previously` means `dynamic` or `static` has been set in previous loop in WHILE of previous call of `nnfw_run()`
bool previously_static(ITensor* op_output) { return ! op_output->is_dynamic(); }

// helper function that check if op's input is static
// Note that input of n'th op has been set to static or dynamic by (n-1)th op.
// That's why it is called `currently_static`
bool currently_static(ITensor* op_input) { return ! op_input->is_dynamic(); }
```

and rewrite the line like the following?

```
if (previously_static(output) && (currently_static(lhs) && currently_static(rhs)))
```

If you have any better idea, it would be awesome.

Reply:
I will do it in the actual PR for merge, thanks!